### DISCUSSION TITLE
v7 pulls in all rows from all collections for a model using getFirstMediaUrl or getMedia

### DISCUSSION THREAD

**Discussion by localpath** [spatie/laravel-medialibrary#2930] [Q&A]
When specifying the collection and conversion to get a url for, the query executed by `HasMediaTrait` seems to get all the items belonging to that model, regardless of specifying the media collection and conversion. It's loading all the models into memory as opposed to just the single model and using it to retrieve the url.

```php
return $this->getFirstMediaUrl('org_logo', 'logo');
```

```php
return $this->getMedia('org_logo')[0]->getUrl('logo');
```

Only one item exists in the media collection for the specified `org_logo` using
```php
$this->addMediaCollection('org_logo')
            ->useFallbackUrl(secure_asset('/img/deka-footer-logo@2x.png'))
            ->singleFile();
```

Any help greatly appreciated!! I supposed I could get the URL myself if needed but wondering if this is a bug or I'm missing something?

![image](https://user-images.githubusercontent.com/44937070/169675019-e94583d1-5585-4c36-a14f-5d9c47679603.png)

![image](https://user-images.githubusercontent.com/44937070/169675025-37fc2b94-a3bf-4ea4-8abf-cd2c5dc32317.png)

**Comment by PaolaRuby** [comment]
V7 has no support

**Reply by localpath** [reply]
Makes sense. For now we're just caching the result "forever" until we upgrade. Do you know if this was a known issue and not in v10?
